presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeWindows.java

```diff
@@ -124,5 +139,15 @@ private static WindowNode collapseWindows(PlanNode source, WindowNode.Specificat
                     canonical.getPrePartitionedInputs(),
                     canonical.getPreSortedOrderPrefix());
         }
+
+        private static boolean dependsOn(WindowNode parent, WindowNode dependent)
+        {
+            return parent.getPartitionBy().stream().anyMatch(symbol -> dependent.getCreatedSymbols().contains(symbol))
```

Comment:
> I'd need to calculate set difference between child's output and child's child output to see what symbols were introduced by the child and see if I depend on any of them.

It is acceptable solution to me, and it will be much better than current one to me. Additionally, `ExpresssionExtractor` has a non recursive version which you may use: `com.facebook.presto.sql.planner.ExpressionExtractor#extractExpressionsNonRecursive`